crates/aptos-telemetry-service/src/auth.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::context::Context;
+use crate::jwt_auth::{authorize_jwt, create_jwt_token, jwt_from_header};
+use crate::types::auth::{AuthRequest, AuthResponse, Claims};
+use aptos_config::config::PeerRole;
+use aptos_crypto::{noise, x25519};
+use aptos_logger::warn;
+use aptos_types::PeerId;
+use warp::filters::BoxedFilter;
+use warp::{
+    filters::header::headers_cloned,
+    http::header::{HeaderMap, HeaderValue},
+    reject, Filter, Rejection,
+};
+use warp::{reply, Reply};
+
+pub fn auth(context: Context) -> BoxedFilter<(impl Reply,)> {
+    warp::path!("auth")
+        .and(warp::post())
+        .and(context.filter())
+        .and(warp::body::json())
+        .and_then(handle_auth)
+        .boxed()
+}
+
+pub async fn handle_auth(
+    context: Context,
+    body: AuthRequest,
+) -> anyhow::Result<impl Reply, Rejection> {
+    let client_init_message = &body.handshake_msg;
+
+    // verify that this is indeed our public key
+    if body.server_public_key != context.noise_config().public_key() {
```

Comment:
What is the server checking here exactly? That the client (validator) included the server's correct Noise public key in the HTTP request?

Reply:
Yes, we are trying to make sure that the client is using the correct public key for the server and inform the client if its using a wrong key.

Reply:
Updated the code comment to be clear.